salvo/src/lib/test_source_manager.py

```diff
@@ -2,26 +2,63 @@
 Test source management operations needed for executing benchmarks
```

Comment:
We may be lacking test coverage, or have too many public methods.

Looks like the code in this PR modifies these public methods:
```python
def determine_envoy_hashes_from_source
def find_all_images_from_specified_tagsdef
def find_all_images_from_specified_sources
def get_envoy_hashes_for_benchmark
def get_image_hashes_from_disk_source
def get_source_repository
def get_source_tree
def get_build_options
def have_build_options
```

However our test file only seems to have tests for:
```python
def test_get_envoy_hashes_for_benchmark
def test_get_image_hashes_from_disk_source
```

I didn't try to generate test coverage, so might be missing something subtle. But if all the other methods need to be public, we should add test cases for them. Ideally each publicly exposed method has a test dedicated to verifying its behaviors. If some of these are just helpers that do get sufficient test coverage from the existing tests here, we can also mark them private. Or use any combination of these approaches.

On a related note, can we add a CI task that will calculate coverage on PRs? This would significantly simplify code reviews. I understand this will take some time and am wondering if you could somehow annotate PRs with the actual coverage until we get there? Alternatively I could run `blaze coverage` locally at code review time.

Reply:
I have a script that generates coverage locally though I haven't integrated that into any CI tasks yet.

Adding more tests is absolutely on our radar.  I can work on adding more tests and update this PR

Reply:
```python
$  egrep "^def test" src/lib/test_source_manager.py | cut -d\( -f 1
def test_get_envoy_hashes_for_benchmark
def test_get_image_hashes_from_disk_source
def test_determine_envoy_hashes_from_source
def test_find_all_images_from_specified_tags
def test_find_all_images_from_specified_sources
def test_get_envoy_hashes_for_benchmark
def test_get_image_hashes_from_disk_source
def test_get_source_tree
def test_get_build_options
def test_have_build_options
```

Reply:
Thank you @abaptiste for increasing the test coverage. Should we file an issue tracking addition of coverage into CI?

While we don't have it in CI, would you mind sharing the script that generates coverage? That will help me to verify coverage as part of the review process.

With that said - if filing of #93 indicates that we don't yet have full coverage for the code in this PR, let's stay here and iterate. We should prefer to submit code together with all its public behaviors tested in a single PR. That way we go from good state to good state on the head. If the scope seems too large for a single PR, we can submit less code, but still with test coverage.

Reply:
@mum4k I added the script I have to measure coverage.  It's messy and still needs work.

If you have a better version/tool, I'm interested.

Using this script, I see the coverage for `source_manager` go from:

```
source_manager.py                         |91.3%    276|    -     0|    -      0
```

To: 
```
source_manager.py                         |97.5%    283|    -     0|    -      0
```

Reply:
Thank you very much for providing the script. I managed to adjust it a bit to generate HTML coverage reports which will help a lot while reviewing the code. We can use it until we implement coverage check in CI.